xrspatial/utils.py

```diff
@@ -98,6 +98,10 @@ def is_dask_cupy(agg: xr.DataArray):
     return isinstance(agg.data, da.Array) and is_cupy_backed(agg)
 
 
+def not_implemented_func(agg, *args):
+    raise NotImplementedError('Not yet implemented.')
```

Comment:
The disadvantage here is that the exception is raised from a different point in the code than where the problem originally occurred. But the stack trace shows this, and the combined use of `not_implemented_func` and `ArrayTypeFunctionMapping` simplifies the code a lot so this is clearly a net improvement.

Reply:
We can add a `messages` param to the `not_implemented_func()` to clearly state the reason if it fails.